test/suite/test_prepare35.py

```diff
@@ -0,0 +1,161 @@
+
+        uri = 'table:test_prepare32'
+        create_params = 'key_format=i,value_format=S'
+        self.session.create(uri, create_params)
+
+        # Step 1: Insert committed baseline data for keys 1-20
+        cursor_committed = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        for i in range(1, 21):
+            cursor_committed.set_key(i)
+            cursor_committed.set_value("committed_value_" + str(i))
+            cursor_committed.insert()
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(25))
+        cursor_committed.close()
+
+        # Step 2: Create first prepared transaction for key 21 with prepared_id=1
+        session_prepare = self.conn.open_session()
+        cursor_prepare = session_prepare.open_cursor(uri)
+        session_prepare.begin_transaction()
+        cursor_prepare.set_key(21)
+        cursor_prepare.set_value("prepared_value_21")
+        cursor_prepare.insert()
+        session_prepare.prepare_transaction('prepare_timestamp=' + self.timestamp_str(30)+',prepared_id=1')
+        cursor_prepare.close()
+
+        # Make stable timestamp equal to prepare timestamp - this should allow checkpoint to reconcile prepared update
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(30))
+
+        # Verify checkpoint writes prepared time window to disk
+        self.checkpoint_and_verify_stats({
+            wiredtiger.stat.conn.rec_time_window_prepared: True,
+        })
+
+        # Step 3: Force eviction to trigger reconciliation with the prepared data
+        # This ensures the prepared update gets written to disk storage
+        session_evict = self.conn.open_session("debug=(release_evict_page=true)")
+        session_evict.begin_transaction("ignore_prepare=true")
+        evict_cursor = session_evict.open_cursor(uri, None, None)
+        for i in range(1, 21):  # Evict committed data pages
+            evict_cursor.set_key(i)
+            self.assertEqual(evict_cursor.search(), 0)
+            evict_cursor.reset()
+        evict_cursor.close()
+        session_evict.rollback_transaction()
+        session_evict.close()
+
+        # Step 4: Rollback the first prepared transaction
+        # This prepends a globally visible tombstone
+        session_prepare.rollback_transaction("rollback_timestamp=" + self.timestamp_str(35))
+        session_prepare.close()
+
+        # Verify key 21 is not found after rollback
+        self.session.begin_transaction('read_timestamp='+ self.timestamp_str(40))
+        read_cursor = self.session.open_cursor(uri, None, None)
+        read_cursor.set_key(21)
+        self.assertEqual(read_cursor.search(), wiredtiger.WT_NOTFOUND)
+        self.session.rollback_transaction()
+
+        # Step 5: Insert second prepared transaction to same key with different prepared_id
+        # This tests the logging code path for handling prepared_id=2
```

Comment:
What is the logging code path?

Reply:
I used to add verbose logging here for debugging, cleaning up the comments in all the tests now